veracode/api.py

```diff
@@ -24,14 +28,16 @@ class VeracodeXMLAPI:
     https://help.veracode.com/reader/LMv_dtSHyb7iIxAQznC~9w/pd_p6JjB9PcDNH3GzWF5Ag
     """
 
-    def __init__(self):
-        # Hard code these to None as they should be specified in the derived classes
-        self._app_id = None
+    def __init__(self, app_name: str):
         self._version = None
 
         ## Use the setter to apply a default to ensure it is valid
         self.base_url = "https://analysiscenter.veracode.com/api/"
 
+        # Set app name and look up ID
+        self._app_name = app_name
+        self._app_id = get_app_id(app_name)
```

Comment:
This means we now globally require on the upload API `getapplist.do` endpoint.  Please document here, in sandbox api, results api, and in markdown.

Reply:
Please document this somewhere

Reply:
Note to self, make a README / wiki edit:
`Require permissions to access UploadAPI along with the other APIs, as get_app_id requires it for app ID resolution.`